### PR TITLE
Add stalin sort algorithm

### PR DESCRIPTION
**Pull request by turtleship69**
### Describe your change:
Implement stalin sort algorithm in pure python in `stalin_sort.py`

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: sorts/stalin_sort.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+Stalin sort iterates over the list, comparing each element with the
+previous one, and removes any that are out of order.
+More information at: https://github.com/gustavo-depaula/stalin-sort
+
+For doctests run following command:
+python -m doctest -v stalin_sort.py
+or
+python3 -m doctest -v stalin_sort.py
+For manual testing run:
+python stalin_sort.py
```
Comment (ZeroDayOwl) [inline sorts/stalin_sort.py:12]:
```suggestion
```

File: sorts/stalin_sort.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+"""
+
+
+def stalin_sort(collection: list) -> list:
+    """Pure python implementation of the stalin sort algorithm in Python
+    :param collection: a list of comparable items
+    :return: all of the items that were in order
```
Comment (ZeroDayOwl) [inline sorts/stalin_sort.py:19]:
```suggestion
    :return: all of the items that were in order
    
```

File: sorts/stalin_sort.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    for i in range(len(collection)):
+        if collection[i - removed] < largest:
+            collection.pop(i - removed)
+            removed += 1
+        else:
+            largest = collection[i - removed]
+    return collection
+
+
+if __name__ == "__main__":
```
Comment (ZeroDayOwl) [inline sorts/stalin_sort.py:41]:
```suggestion
if __name__ == "__main__":
    import doctest
    doctest.testmod()
```

File: sorts/stalin_sort.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    user_input = input("Enter numbers separated by a comma:\n").strip()
+    unsorted = [int(item) for item in user_input.split(",")]
+    print(stalin_sort(unsorted))
```
Comment (ZeroDayOwl) [inline sorts/stalin_sort.py:44]:
```suggestion
```

**Review by ZeroDayOwl — CHANGES_REQUESTED** [review]

File: sorts/stalin_sort.py (hunk 4 above)
Comment (ZeroDayOwl) [reply sorts/stalin_sort.py:44]: Hey @turtleship69 you don't need to use `input()` or `print()` function for the algorithm.

**Review by CaedenPH — COMMENTED** [review]
Small changes

**Review by CaedenPH — APPROVED** [review]
@cclauss Can you review please

File: sorts/stalin_sort.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
+"""
+This is a pure Python implementation of the stalin sort algorithm.
+Stalin sort iterates over the list, comparing each element with the
+previous one, and removes any that are out of order.
+More information at: https://github.com/gustavo-depaula/stalin-sort
+"""
+
+
+def stalin_sort(collection: list[int]) -> list[int]:
+    """Pure python implementation of the stalin sort algorithm in Python
```
Comment (cclauss) [inline sorts/stalin_sort.py:10]:
```suggestion
    """Pure Python implementation of the stalin sort algorithm
```

File: sorts/stalin_sort.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+    Examples:
+    >>> stalin_sort([0, 5, 3, 2, 2])
+    [0, 5]
+    >>> stalin_sort([])
+    []
+    >>> stalin_sort([-2, -5, -45])
+    [-2]
+    """
+    if collection == []:
```
Comment (cclauss) [inline sorts/stalin_sort.py:22]: All other falsy values are just as dangerous as an empty list.
```suggestion
    if not collection:
```

**Comment by CaedenPH** [comment]
@cclauss This is ready for merge

**Comment by cclauss** [comment]
This is a useless algorithm that destroys data.  I believe that there is enough of a shrine to this silliness at https://github.com/gustavo-depaula/stalin-sort

**Pull request closed without merging**